### DISCUSSION TITLE
How to import direct stage-damage functions

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#336] [Q&A]
We need to figure out the best way, if there is a way, to import direct stage-damage functions. 

### 1.4.3 Background
Instead of relying on a depth-percent damage function to calculate dollar damages at a given structure, the user has the option of directly specifying a stage-damage function for that structure. This direct stage-damage function is entered in the user interface as part of the structure inventory, but the direct stage-damage function is exported as part of the occupancy type data. 

### 2.0 import process
An HEC-FDA Version 2.0 study can be created using HEC-FDA Version 1.4.3 study data. H&H input functions, levees, and occupancy types are imported. The user then needs to minimally import an impact area set before importing aggregated stage-damage functions from 1.4.3. Then, the import procedure groups stage-damage functions by plan and year to create a set of stage damage functions with the same plan and year, in which each stage-damage function is assigned to its particular impact area, damage category, and asset category. 

### The problem 
Direct stage-damage functions are not being imported into the correct place. The functions are currently being imported as depth-percent damage functions that live in their own "occupancy type" objects, while the direct stage-damage functions _should_ be part of each relevant set of stage-damage functions as manually-entered stage-damage functions). 

### Complications
- The assignment of a direct stage-damage function in HEC-FDA Version 1.4.3 to a particular plan and year is controlled through the use of structure modules. The assignment of a structure to a particular module can be found among the structure inventory. The assignment of a module to a particular plan and year can be obtained in the 1.4.3 exported study data. 
 
### Show and tell
- In the image below, you'll see an example direct stage-damage function as part of the block of occupancy type data. The only relational data is the structure ID - 362 - part of the damage function name "Direct000362"
- below the direct stage damage function is the list of modules
- below the list of modules is the plan-year-module relations 

![image](https://user-images.githubusercontent.com/25163165/166828102-0d4bff5a-1f2e-47ff-aa26-6ffe730cfe58.png)

### Potential solutions
The best I've come up with is to not import the direct stage-damage functions and ask the user to instead manually enter them. It seems rather complicated but not impossible to import direct stage-damage functions together with the aggregated stage-damage functions to correctly create each set of stage-damage functions. We'd have to import occupancy type and aggregated stage-damage data together, then look for the structure to which a direct stage-damage function belongs, then look for the module to which a structure has been assigned, then look for the plan and year to which the module has been assigned, and put the direct stage-damage relationship with each set of stage-damage functions for each plan-year combination to which the module (<---- structure <---- direct stage-damage) has been assigned. Complicated but not impossible. Thoughts?

**Comment by HenryGeorgist** [comment]
what is the value of the direct damage function? Having the user convert the damage function to a percent damage and assign the appropriate occupancy type to the appropriate structure (and structure value) seems like  a reasonable solution to me. 

It will put a burden on the user. How much is this feature used typically? And Why?

**Reply by rnugent3** [reply]
having the user convert the stage-damage function into a depth-percent damage function and then creating a unique occupancy type seems like more work than just adding a manually-entered stage-damage function to each relevant set of stage-damage functions. 

This is often used for unique commercial structures.

**Reply by rnugent3** [reply]
I don't have data on how often this used. Anecdotally, I have seen direct damage functions in several hec-fda studies, out of however many I've seen. I think these are probably used more often than fragility functions or exterior-interior functions - I've only seen each of these once.

**Comment by rnugent3** [comment]
Hi All! I've been thinking a little further on this topic. We could simply ask the user to either: (1)  The user could use distinct damage categories for the structures with direct damage functions and then paste in the direct damage function for the damage category specific to that structure.  (2) If there is desire to use a common damage category, the user could add the means and the variances of the relevant functions. This works now because we'll allow editing of the computed stage-damage functions. 

Thoughts?

**Reply by rnugent3** [reply]
This is related to Issue #338
